core/src/main/java/tachyon/master/TachyonMaster.java

```diff
@@ -188,6 +188,9 @@ private boolean isFormatted(String folder, String path) throws IOException {
       folder += TachyonURI.SEPARATOR;
     }
     UnderFileSystem ufs = UnderFileSystem.get(folder, mTachyonConf);
+    if (ufs.providesStorage() == false) {
+      return true;
+    }
```

Comment:
Instead of having the check here, could it be checked before this method is called? (ie don't do the preconditions check unless the underfilesystem provides storage).